addons/mail/static/src/discuss/core/common/notification_settings.js

```diff
@@ -0,0 +1,98 @@
+/* @odoo-module */
+
+import { useState, Component } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const notificationTypeRegistry = registry.category("mail.discuss/notification_types");
+notificationTypeRegistry
+    .add("all", {
+        name: _t("All Messages"),
+    })
+    .add("mentions", {
+        name: _t("Mentions Only"),
+    })
+    .add("none", {
+        name: _t("Nothing"),
+    });
+const muteDurationRegistry = registry.category("mail.discuss/mute_durations");
+muteDurationRegistry
+    .add("15_mins", {
+        value: 15,
+        name: _t("For 15 minutes"),
+    })
+    .add("1_hour", {
+        value: 60,
+        name: _t("For 1 hour"),
+    })
+    .add("3_hours", {
+        value: 180,
+        name: _t("For 3 hours"),
+    })
+    .add("8_hours", {
+        value: 480,
+        name: _t("For 8 hours"),
+    })
+    .add("24_hours", {
+        value: 1440,
+        name: _t("For 24 hours"),
+    })
+    .add("forever", {
+        value: -1,
+        name: _t("Until I turn it back on"),
+    });
+
+export class NotificationSettings extends Component {
+    static components = { Dropdown, DropdownItem };
+    static props = ["hasSizeConstraints?", "thread", "close", "className?"];
+    static template = "discuss.NotificationSettings";
+
+    setup() {
+        this.threadService = useState(useService("mail.thread"));
+    }
+
+    get settingItems() {
+        return notificationTypeRegistry.getEntries().map(([id, item]) => ({
+            id,
+            name: item.name,
+            active: this.props.thread.notificationType === id,
+        }));
+    }
```

Comment:
Same  remark: no need this getter with reformatting registry stuff.
For the `active`, you can have a dedicated function for that

Reply:
same comment as previous